File: Minimizing_Coins.py
n, x = map(int, input().split())
c = list(map(int, input().split()))
dp = [1 << 60 for _ in range(x + 1)]
dp[0] = 0
for i in range(1, x + 1):
    chck = [dp[i - j] + 1 for j in c if i >= j]
    chck = min(chck) if chck else 1 << 60
    dp[i] = min(dp[i], chck)
print(dp[x] if dp[x] != 1 << 60 else -1)

File: Minimizing_Coins.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, x;
    cin >> n >> x;
    vector<int> c(n);
    for (int &ci: c) {
        cin >> ci;
    }
    vector<long long> dp(x + 1, 1LL << 60);
    dp[0] = 0LL;
    for (int i = 1; i <= x; i++) {
        vector<long long> chck;
        for (int j: c) {
            if (i >= j) {
                chck.push_back(dp[i - j] + 1LL);
            }
        }
        long long val = 1LL << 60;
        if ((int)chck.size()) {
            val = *min_element(chck.begin(), chck.end());
        }
        dp[i] = min(dp[i], val);
    }
    if (dp[x] != 1LL << 60) {
        cout << dp[x];
    } else {
        cout << -1;
    }
}